scalableminds/webknossos
# Add Quick Select Tool

Adds a quick select tool which allows the user to draw a rectangle around a segment. Then, the tool will try to fill the segment beginning from the center using the intensity values of the visible color layer. The tool picks a threshold for the flood fill so that the borders are ideally not touched by the flood fill.

The tool has an own settings menu with which one can control some parameters. Especially, the preview mode is interesting, since it enables the user to fine tune values while getting instant feedback.

Additionally, this PR upgrades antd by a patch version, since this allowed to control the visibility of the settings popover for the new tool.

### Known Issues
- [x] ~for RGB data, the preview doesn't render for some reason (any ideas?)~ the problem was the clipping distance. I used the kiwi dataset which has a huge scale (this list even included this issue, but I still didn't understand that initially :see_no_evil:)
- ~[ ] in coarser mags, the fill hole operation doesn't seem to work sometimes (don't know why this doesn't happen in mag 1)~ cannot repro anymore
- [x] the preview is tied to the clipping distance which can give unexpected experiences (e.g., when moving in z, the preview can stay there even though one would expect it to disappear)
- [x] the preview mode is not super easy to distinguish from actually labeled data. ~I tried to render a rectangular border around the preview, but ran into quite a bit of obstacles which is why I aborted the idea. See 7db28bfca.~ ~I also tried to use 4 different plane geometries to paint the borders (see c349f9f43), but this turned out to be quite finicky as well (the combination of rotated geometry, dimension transpositions for the viewports and the dataset scale didn't want to work together).~ In the end, I simply reused the preview texture to paint the border voxels there. If the dataset is anisotropic, the border will be, too, but I think, it still feels ok, because it's in line with how the voxels look anyway.

### URL of deployed dev instance (used for testing):
- https://magicwandheuristic.webknossos.xyz

### Steps to test:
- create a new volume annotation (ideally, without fallback layer)
- use the new tool for some segments (dark and bright)
- use the settings
- check for intuitiveness
- do some monkey testing to provoke bugs (mistaken heuristics don't count :stuck_out_tongue:)

### Issue
- fixes #6469 

------
(Please delete unneeded items, merge only when none are left open)
